spring-web/src/main/java/org/springframework/http/converter/ByteArrayHttpMessageConverter.java

```diff
@@ -53,11 +52,7 @@ public boolean supports(Class<?> clazz) {
 
 	@Override
 	public byte[] readInternal(Class<? extends byte[]> clazz, HttpInputMessage inputMessage) throws IOException {
-		long contentLength = inputMessage.getHeaders().getContentLength();
-		ByteArrayOutputStream bos =
-				new ByteArrayOutputStream(contentLength >= 0 ? (int) contentLength : StreamUtils.BUFFER_SIZE);
-		StreamUtils.copy(inputMessage.getBody(), bos);
-		return bos.toByteArray();
+		return inputMessage.getBody().readAllBytes();
```

Comment:
The same comment made above applies here, but this commit also removes the `contentLength` optimisation. I think using `InputStream::readNBytes(contentLength >= 0 ? (int) contentLength : StreamUtils.BUFFER_SIZE)` makes more sense.